services/EventAPIService.js

```diff
@@ -0,0 +1,76 @@
+const axios = require("axios");
+const { API_100MS_BASE_URL } = require("../constants/events");
+
+// A service class for all REST API operations
+class EventAPIService {
```

Comment:
Is there any specific reason for using the `class-based` Approach?

Reply:
Hey @SanketDhabarde,
Thank you for pointing this out.

I have tried implementing the same logic with function based approach but here are few reasons to use class based approach:

-  The `EventAPIService`  class encapsulates the Axios instance, token service instance, and the configuration logic within a single unit  to organize and modularize the code, to make it easy to understand and maintain. 
- The `EventAPIService` class uses private class fields `#axiosInstance` and `#tokenServiceInstance` to ensures that these variables are only accessible within the class itself, preventing unintended modifications or access from outside the class. 
- For using the interceptors, the class encapsulates the Axios interceptors and request/response handling logic. The centralized approach ensures consistent handling of requests and responses.
- The `EventTokenService` class constructor initializes the `managementToken` by calling the `getManagementToken` method with the argument `true`. This ensures that the `managementToken` is set during object creation, providing a default value for the instance. 

Apart from that, I had read in the documentation to go for the class based approach and in all the starter template and example code given by 100ms they are also using the class based approach.